Ledger Digest Upload SDK

rewongmicrosoft opened the pull request:
Swagger PR: https://github.com/Azure/azure-rest-api-specs/pull/13871, https://github.com/Azure/azure-rest-api-specs/pull/13916/files

# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/master/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running `generate.ps1`. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version.
	
	**Note: We have recently updated the PSH module called by `generate.ps1` to emit additional data. This would help reduce/eliminate the Code Verification check error. Please run following command**:

	    `dotnet msbuild eng/mgmt.proj /t:Util /p:UtilityName=InstallPsModules`

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

rewongmicrosoft committed:
adding ledger changes

rewongmicrosoft committed:
adding ledger changes

rewongmicrosoft committed:
all generated changes

rewongmicrosoft committed:
all generated changes

rewongmicrosoft committed:
Merge branch 'rewong_ledger' of https://github.com/rewongmicrosoft/azure-sdk-for-net into rewong_ledger

rewongmicrosoft committed:
generated files updates

rewongmicrosoft committed:
adding tests, new generated files from databases

rewongmicrosoft committed:
updating recorded tests for ledger

markcowl commented on sdk/sqlmanagement/Microsoft.Azure.Management.Sql/tests/LedgerDigestUploadsTests.cs:
```diff
@@ -0,0 +1,90 @@
+                // Get the created database and validate IsLedgerOn parameter is set
+                Database databaseResponse = sqlClient.Databases.Get(resourceGroup.Name, server.Name, databaseName);
+                Assert.True(databaseResponse.IsLedgerOn);
+            }
```
The test failure you are seeing is caused by a mismatch between the class name for your test path and the path segment in the recording files:  LedgerDigestUploadsTests (path segment) -> LedgerDigestUploadTests (actual class name)

rewongmicrosoft replied:
thank you, will fix this

markcowl commented on sdk/sqlmanagement/Microsoft.Azure.Management.Sql/tests/LedgerDigestUploadsTests.cs:
```diff
@@ -0,0 +1,90 @@
+                Database databaseResponse = sqlClient.Databases.Get(resourceGroup.Name, server.Name, databaseName);
+                Assert.True(databaseResponse.IsLedgerOn);
+            }
+        }
```
The test failure you are seeing is caused by a mismatch between the class name for your test path and the path segment in the recording files: LedgerDigestUploadsTests (path segment) -> LedgerDigestUploadTests (actual class name)

markcowl reviewed (changes requested):
One comment pertains to your current build failure.  You will also need to make sure that this file is updated to reflect the latest commit and tag from the swagger repo: https://github.com/Azure/azure-sdk-for-net/blob/master/eng/mgmt/mgmtmetadata/sql_resource-manager.txt 

Normally the 'generate.ps1' script would take care of this for you.

markcowl commented:
@rewongmicrosoft Reviewed, please update.

rewongmicrosoft committed:
renaming file path

rewongmicrosoft committed:
adding ListByServerWithHttpMessagesAsync to database operations customizations to fix breaking change

rewongmicrosoft commented on sdk/sqlmanagement/Microsoft.Azure.Management.Sql/src/Generated/DatabasesOperations.cs:
```diff
@@ -501,7 +503,7 @@ public async Task<AzureOperationResponse<IEnumerable<MetricDefinition>>> ListMet
         /// A response object containing the response body and response headers.
         /// </return>
-        public async Task<AzureOperationResponse<IPage<Database>>> ListByServerWithHttpMessagesAsync(string resourceGroupName, string serverName, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken))
+        public async Task<AzureOperationResponse<IPage<Database>>> ListByServerWithHttpMessagesAsync(string resourceGroupName, string serverName, string skipToken = default(string), Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken))
```
thanks, added the old signature to the customizations folder in the existing partial class there. Checking with the owner of this code change in the SQL codebase to make sure it looks ok

rewongmicrosoft committed:
fixing newline spacing

rewongmicrosoft committed:
upating based on most recent generation

rewongmicrosoft committed:
adding sql_resource-manager.txt file

rewongmicrosoft commented:
@markcowl addressed your comments and checks are passing, still waiting on SQL engineer to validate fix for breaking change is valid. Once we get confirmation from him, we are ok to merge

markcowl commented on sdk/sqlmanagement/Microsoft.Azure.Management.Sql/src/Customizations/DatabasesOperations.cs:
```diff
@@ -61,5 +63,50 @@ internal partial class DatabasesOperations
                 customHeaders,
                 cancellationToken).ConfigureAwait(false);
         }
+
```
This looks good, you would need similar customizations in the partial interface for IDatabasesOperations and the ListByServerAsync method in the DatabasesOperationsExtensions partial class [although, since this is an extension method, it doden't need to reside in the same static class]

rewongmicrosoft replied:
ok I understand now. Made the changes to add the interface in IDatabasesOperations, and add ListByServerAsync without skip token in the DatabasesOperationsExtensions class

markcowl commented:
@rewongmicrosoft updated my review - this also requires customizations in the interface and extensions (essentially analogous to this one).

rewongmicrosoft committed:
final updates to ListByServerWithHttpMessagesAsync interface and caller

rewongmicrosoft commented:
@markcowl made updates, we are good to check in as soon are you approve

rewongmicrosoft commented:
/azp run net - mgmt

ericshape commented:
/azp run net - mgmt

xaliciayang commented:
/azp run net - mgmt

xaliciayang commented:
/azp run
